core/services/ardupilot_manager/ArduPilotManager.py

```diff
@@ -112,74 +112,69 @@ def restart(self) -> bool:
 
     def _load_endpoints(self) -> None:
         """Load endpoints from the configuration file to the mavlink manager."""
-        if "endpoints" not in self.configuration:
-            self.configuration["endpoints"] = []
-        endpoints = self.configuration["endpoints"]
+        endpoints = {Endpoint(**endpoint) for endpoint in self.configuration.get("endpoints") or []}
         for endpoint in endpoints:
             try:
-                if not self.mavlink_manager.add_endpoint(Endpoint(**endpoint)):
+                if not self.mavlink_manager.add_endpoint(endpoint):
                     raise ValueError("Cannot add endpoint.")
             except Exception as error:
                 warn(f"Could not load endpoint {endpoint}. {error}")
 
-    def _reset_endpoints(self, endpoints: List[Endpoint]) -> None:
+    def _reset_endpoints(self, endpoints: Set[Endpoint]) -> None:
         try:
             self.mavlink_manager.clear_endpoints()
             self.mavlink_manager.add_endpoints(endpoints)
             print("Reseting endpoints to previous state.")
         except Exception as error:
             warn(f"Error reseting endpoints: {error}")
 
-    def _update_endpoints(self, updated_endpoints: List[Endpoint]) -> bool:
+    def _update_endpoints(self) -> bool:
         try:
-            self.configuration["endpoints"] = deepcopy(updated_endpoints)
+            self.configuration["endpoints"] = [endpoint.asdict() for endpoint in self.get_endpoints()]
```

Comment:
Is there a problem to have the configuration as a set ?

Reply:
Yes. JSON does not support sets.